compiler/testData/lineNumber/defaultParameter.kt

```diff
@@ -1,4 +0,0 @@
-class A {
-    fun foo(param: Int = test.lineNumber()) {
```

Comment:
Converted test doesn't test line number on default parameter initialization. It possible to make breakpoint on this line and this behaviour should be somehow tested (maybe in debugger tests if they doesn't have such test yet)

Reply:
Yikes, thanks for catching that Mike!

The issue is that foo$default is synthetic and I decided to skip synthetic methods so that the stepping tests operate as going through the code as step-into would. However, if you just do step-into, you do not get the step in the synthetic foo$default method and therefore, we do not check that you can set a breakpoint there.

I have updated the pull request to add a flag for requesting force-step-into behavior where we see the synthetic methods. I have used that flag in the two tests where it makes sense: the two tests with $default methods.

What do you think?

Reply:
Thanks! Look OK!